python/cuml/test/test_array.py

```diff
@@ -480,6 +535,43 @@ def test_cumlary_binops(operation):
     assert(cp.all(ary_c.to_output('cupy') == c))
 
 
+def test_sliced_array_owner():
+    """
+    When slicing a CumlArray, a new object can be created created which
+    previously had an incorrect owner. This was due to the requirement by
+    `cudf.core.Buffer` that all data be in "u1" form. CumlArray would satisfy
+    this requirement by calling
+    `cp.asarray(data).ravel(order='A').view('u1')`. If the slice is not
+    contiguous, this would create an intermediate object with no references
+    that would be cleaned up by GC causing an error when using the memory
+    """
+
+    # Create 2 copies of a random array
+    random_cp = cp.array(cp.random.random((500, 4)), dtype=np.float32)
+    cupy_array = cp.array(random_cp, copy=True)
+    cuml_array = CumlArray(random_cp)
+
+    # Make sure we have 2 pieces of data
+    assert cupy_array.data.ptr != cuml_array.ptr
+
+    # Since these are C arrays, slice off the first column to ensure they are
+    # non-contiguous
```

Comment:
would be nice to add test for F arrays as well, just for completeness

Reply:
Added F test.